### PR TITLE
Change-Notification nodejs sample.

### PR DESCRIPTION
**Pull request by Sudhir-MSFT**
Description
This sample shows a status of a user in personal scope and demonstrates presence subscription lifecycle followed by commands given to Bot

Language Used: JavaScript/nodejs

Proposed features
User Subscription, Bot, Status Change Notification, Bot SSO

Testing
After successfully installation of app you will get a sign in button. When sign in is complete then you get your current status in adaptive card
image
image

- After that when the user status changes you will get notify about their status: 
- Change user status from available to busy like
image
- Change user status from busy to busy offline
 image

### CODE REVIEW

File: samples/graph-change-notification/nodejs/Helper/SubscriptionManager.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    getGraphClient() {
+        const client = Client.init({
+            authProvider: (done) => {
+                done(null, this.accessToken);
+            }
+        });
+        return client;
+    }
+    async deleteSubscription(subscriptionId) {
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/Helper/SubscriptionManager.js:18]: Provide one line space

File: samples/graph-change-notification/nodejs/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+  A) Or from Visual Studio code
+
+  - Launch Visual Studio code
+  - File -> Open Folder
+  - Navigate to `samples/graph-change-notification/nodejs` folder
+  - Select `ChangeNotification` folder
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/README.md:25]: Remove ChangeNotification folder line

File: samples/graph-change-notification/nodejs/bots/dialogBot.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                            },
+                            {
+                                "title": "Activity:",
+                                "value": activity
+                            }
+                        ]
+                    }
+                ]
+            }
+            var tdata = CardFactory.adaptiveCard(card);
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/bots/dialogBot.js:79]: use const

File: samples/graph-change-notification/nodejs/deploymentTemplates/template-with-new-rg.json [hunk 4]
```diff
@@ -0,0 +1,183 @@
+{
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/deploymentTemplates/template-with-new-rg.json:1]: is this file required? Please check

File: samples/graph-change-notification/nodejs/deploymentTemplates/template-with-preexisting-rg.json [hunk 5]
```diff
@@ -0,0 +1,154 @@
+{
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/deploymentTemplates/template-with-preexisting-rg.json:1]: Check this file as well if not required please remove

File: samples/graph-change-notification/nodejs/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        console.log("In Response");
+        clientStatesValid = false;
+        console.log(req.body.value[0].resourceData);
+        let userstatus = req.body.value[0].resourceData;
+        console.log(userstatus.activity);
+        console.log(userstatus.availability);
+        status = 202;
+        //for storing step context
+        const dbot = new DialogBot(conversationState, userState, dialog, conversationReferences);
+        let _notication = "Change your status to get notification";
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/index.js:102]: const

File: samples/graph-change-notification/nodejs/package.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/Microsoft/BotBuilder-Samples.git"
+    },
+    "dependencies": {
+        "@microsoft/microsoft-graph-client": "^2.2.1",
+        "axios": "^0.21.1",
+        "botbuilder": "~4.11.0",
```
Comment (Trinetra-MSFT) [inline samples/graph-change-notification/nodejs/package.json:21]: Update the bot builder package to latest check other packages as well

**Review by Trinetra-MSFT — CHANGES_REQUESTED** [review]

**Review by Trinetra-MSFT — APPROVED** [review]

File: samples/graph-change-notification/nodejs/constant.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+exports.subscriptionConfiguration = {
+    changeType: 'updated',
+    notificationUrl: 'https://*******.ngrok.io/api/notifications',
```
Comment (Wajeed-msft) [inline samples/graph-change-notification/nodejs/constant.js:3]: Could you please pull this from config file (.env)?

File: samples/graph-change-notification/nodejs/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+Bot Framework v4 ChangeNotification sample.
+
+This sample app demonstrates sending change notifications to user presence in Teams based on user presence status.
+
+
+## Prerequisites
+
+- Microsoft Teams is installed and you have an account
+- [.NET Core SDK](https://dotnet.microsoft.com/download) version 3.1
```
Comment (Wajeed-msft) [inline samples/graph-change-notification/nodejs/README.md:11]: Could you please remove this and update steps specific to node.js enviroment?

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/graph-change-notification/nodejs/README.md (hunk 9 above)
Comment (Sudhir-MSFT) [reply samples/graph-change-notification/nodejs/README.md]: Yes

**Review by Wajeed-msft — APPROVED** [review]
